File: 백준/Bronze/10804. 카드 역배치/카드 역배치.swift
import Foundation

var nums:[Int] = []
for n in 1...20 { nums.append(n) }

for _ in 0..<10 {
    var ab = readLine()!.split(separator: " ").map{Int(String($0))!}
    ab[0] -= 1
    ab[1] -= 1
    let m = (ab[1]-ab[0])/2 + 1
    for i in 0..<m {
        let temp = nums[ab[0]+i]
        nums[ab[0]+i] = nums[ab[1]-i]
        nums[ab[1]-i] = temp
    }
}

var answer = ""
for n in nums { answer += "\(n) "}
print(answer)

File: 백준/Bronze/10804. 카드 역배치/카드 역배치.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] nums = new int[20];
        for(int i=0; i<20; i++) nums[i] = i+1;

        for(int i=0; i<10; i++) {
            String s = br.readLine();
            StringTokenizer st = new StringTokenizer(s);
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int m = (b-a)/2 + 1;

            for(int j=0; j<m; j++) {
                int temp = nums[a-1+j];
                nums[a-1+j] = nums[b-1-j];
                nums[b-1-j] = temp;
            }
        }

        String answer = "";
        for(int i=0; i<20; i++) answer += nums[i] + " ";
        System.out.println(answer);
    }
}